### ISSUE TITLE
O2 decode equivalence does not hold between the TP verify rows and the PP replay: one DECIDED argmax flip at round 18, and the honest band is 5e-1

### ISSUE THREAD

**Issue by avifenesh** [avifenesh/memra#487]
## Claim under test

The O2 oracle asserts decode equivalence between the **TP verify rows** (the truth: a batched
verify over a symmetric TP2 topology) and the **PP serial replay** of the same rows. With the
band repaired by #479, that claim does not hold at this depth. This is a finding against the PP
replay path, not a band-calibration request. **Nothing was widened to make a cell pass, and
nothing should be.**

## Why this is visible only now

Before #479 the O2 band was an elementwise **relative** bound of 2e-4 on **raw logits** across a
TP all-reduce versus a PP serial walk, which is unachievable by construction. It aborted the pp
arm at `round=0 row=0 delta=0.03362177 limit=0.0002`, so rounds 1..51 had never executed and the
disagreement below had never been measured. #479 made the bound **absolute in probability space**
(`TP_SPEC_O2_PROB_BAND`), censused every replayed row, and judged after the walk instead of
aborting inside it. The gate did not get looser; it got honest, and the honest gate exposed this.

## The measurement

Full 52-round walk on the repaired engine, 2 x B200 (sm_100a), engine test binary
`memra-engine-o2band-b` sha256 `2a89eed5...`, fingerprint `memra-0.136.0-e9541cb99513`,
census band 0.5 (deliberately wide: this run measures the distribution, it is not a calibrated
number). Wall 278.76 s, exit 101, panic at
`crates/memra-engine/src/glm5_tp_spec_pair_gpu.rs:341`.

**213 rows over all 52 rounds. 5 argmax flips.**

Probability-space `max_dp` distribution over all 213 rows:

| stat | value |
|---|---|
| min | 6.7900647e-10 |
| median | 0.003537449066 |
| p95 | 0.1063528087 |
| p99 | 0.1863231802 |
| max | 0.580590269 |
| rows over 1e-4 / 1e-3 / 1e-2 / 1e-1 | 144 / 121 / 82 / 12 |

`max_dp` over the rows where both routes **agree** on argmax: **0.2240724683**.

### The decided row

Round 18 row 1. The replay picks token **11** with a **0.710** probability gap over the truth's
pick; the truth picks token **10989** with a **0.429** gap the other way. `max_dp` 0.581. Both
routes are confident and they are confident about different tokens, so this is not a tie that any
band explains: it is a decode divergence.

```
round  row  max_dp        pick_replay  pick_truth  gap_replay    gap_truth
18     1    5.805902690e-1  11         10989       7.100184314e-1  4.287556295e-1
```

All five flips:

| round | row | max_dp | pick_replay | pick_truth | gap_replay | gap_truth |
|---|---|---|---|---|---|---|
| 11 | 1 | 0.07188638042 | 320 | 22840 | 0.07488191934 | 0.06068156633 |
| 18 | 1 | 0.580590269 | 11 | 10989 | 0.7100184314 | 0.4287556295 |
| 41 | 2 | 0.02434970007 | 7981 | 1614 | 0.04472779847 | 0.003961147763 |
| 44 | 0 | 0.07288647818 | 13473 | 3175 | 0.007209048208 | 0.1279578251 |
| 46 | 0 | 0.1642458379 | 9955 | 11206 | 0.2466308057 | 0.08184676427 |

## Why no band fixes this

The lane's band rule is "twice the worst agreeing row, rounded to {1,2,5}x10^k". On this census
that returns **5e-1**, a bound covering half the probability simplex. That number is honest and it
is useless as a gate, which is the point: a band that admits round 18 is not a decode-equivalence
check any more. Do not tune the band to make this green.

## What is NOT implicated

- **TP is clean at the stronger claim.** The tp arm passes O3:
  `O3 PASS: 160 teacher-forced tokens, widths 1..7, every logit bit-identical` (the arm #387
  needs). The plain arm passes O1.
- The judge itself is exercised by #479's red arm (11 CPU cases).

So the disagreement is between the PP serial replay and the TP verify rows, and the TP side is
bit-exact against its own teacher-forced tape.

## Receipts

On the dev pair at `/root/cell12/receipts/pp-census-3/` (`summary.json`, `oracle-pp.log`,
`env-pp.txt`, `oracles/pp/o2-prob-band.tsv`). Banked off-box in darklanes (private)
`research/glm5-o2-band-20260911/cell12b/`: the full 213-row TSV with both gaps per row, the
pinned pp env, the oracle log, and the summary. The box is rented, so read the banked copy.

## Asks for the PP replay owner

1. Decide whether the PP replay is expected to reproduce the TP verify rows' argmax at this depth
   at all. If it is not, the O2 oracle's pp arm is claiming something the engine never promised
   and the claim should be narrowed in the oracle, explicitly and in writing, not by a band.
2. If it is expected, round 18 row 1 is the reproducer: two confident routes, different tokens.
3. Until one of those lands, the pp **oracle** arm stays out of the pinned cell (darklanes cell 14
   runs plain O1 + tp O3 + the full served phase; the pp **server** arm still runs, because a
   serving posture is not an oracle claim).
